src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/writers/CommonJSONWriter.java

```diff
@@ -257,4 +268,14 @@ public void writeEndObject() throws IOException {
     public void close() throws IOException {
         generator.close();
     }
+
+    protected boolean skipObjectWriting(EncodingHints encodingHints) {
+        Boolean skipIfSingleFeature =
+                getEncodingHintIfPresent(encodingHints, SKIP_OBJECT_ENCODING, Boolean.class);
+        return skipIfSingleFeature != null
+                && skipIfSingleFeature.booleanValue()
+                && isSingleFeatureRequest()
+                && (identifier.equals(TemplateIdentifier.GEOJSON)
```

Comment:
Since this is a "common json writer" what else can it be... I see ``JSON("application/json", "geojson-template.json")`` in the list, what is that for?

Reply:
There are two JSON indentifier one for geo+json and one for application/json. The CommonJsonWriter works for both and the check is meant to encode the output as a single feature only in case the mime is geo+json as the OGCAPI already does.